File: Lab6: Auth+JWT/frontend/src/types/pages.ts
export interface Page {
    id: string,
    title: string,
    body: string,
}

export interface PageOverview extends Omit<Page, 'body'> {}

export interface PageStas extends Omit<Page, 'body'> {
    views: number,
    time_spent: number,
    last_visit: number, // Дату питона мы перевели в Unix Time
}

File: Lab6: Auth+JWT/backend/schemas/pages.py
from typing import Optional
from pydantic import BaseModel
from datetime import datetime


class Page(BaseModel):
    id: str = ''
    title: str = 'Страница'
    body: str = '<main>Здесь ничего нет.</main>'

class Kpi(BaseModel):
    id: int = 0
    page_id: str = ''
    views: int = 0
    time_spent: int = 0
    last_visit: Optional[datetime] = None

class PageTime(BaseModel):
    time_spent: int = 0

